utils/db.ts

```diff
@@ -422,106 +422,32 @@ export async function incrVisitsCountByDay(date: Date) {
-  return await getValue<bigint>([
-    "items_count",
-    formatDate(date),
-  ]);
-}
-
-export async function getVotesCountByDay(date: Date) {
-  return await getValue<bigint>([
-    "votes_count",
-    formatDate(date),
-  ]);
-}
-
-export async function getUsersCountByDay(date: Date) {
-  return await getValue<bigint>([
-    "users_count",
-    formatDate(date),
-  ]);
-}
-
-export async function getAllVisitsCountByDay(options?: Deno.KvListOptions) {
-  const iter = await kv.list<bigint>({ prefix: ["visits_count"] }, options);
-  const visits = [];
+/** Gets all dates since a given number of milliseconds ago */
+export function getDatesSince(msAgo: number) {
   const dates = [];
-  for await (const res of iter) {
-    visits.push(Number(res.value));
-    dates.push(String(res.key[1]));
-  }
-  return { visits, dates };
-}
-
-export async function getAllItemsCountByDay(options?: Deno.KvListOptions) {
-  const iter = await kv.list<bigint>({ prefix: ["items_count"] }, options);
-  const visits = [];
-  const dates = [];
-  for await (const res of iter) {
-    visits.push(Number(res.value));
-    dates.push(String(res.key[1]));
-  }
-  return { visits, dates };
-}
-
-export async function getAllVotesCountByDay(options?: Deno.KvListOptions) {
-  const iter = await kv.list<bigint>({ prefix: ["votes_count"] }, options);
-  const visits = [];
-  const dates = [];
-  for await (const res of iter) {
-    visits.push(Number(res.value));
-    dates.push(String(res.key[1]));
-  }
-  return { visits, dates };
-}
+  const now = Date.now();
+  const start = new Date(now - msAgo);
 
-export async function getAllUsersCountByDay(options?: Deno.KvListOptions) {
-  const iter = await kv.list<bigint>({ prefix: ["users_count"] }, options);
-  const visits = [];
-  const dates = [];
-  for await (const res of iter) {
-    visits.push(Number(res.value));
-    dates.push(String(res.key[1]));
+  while (+start < now) {
+    start.setDate(start.getDate() + 1);
+    dates.push(new Date(start));
   }
-  return { visits, dates };
-}
 
-export async function getAnalyticsMetricListPerDay(
-  metric: string,
-  options?: Deno.KvListOptions,
-) {
-  const iter = await kv.list<bigint>({ prefix: [metric] }, options);
-  const metricsValue = [];
-  const dates = [];
-  for await (const res of iter) {
-    metricsValue.push(Number(res.value));
-    dates.push(String(res.key[1]));
-  }
-  return { metricsValue, dates };
+  return dates;
 }
 
-export async function getManyAnalyticsMetricsPerDay(
-  metrics: string[],
-  options?: Deno.KvListOptions,
+export async function getManyMetrics(
+  metric: "visits_count" | "items_count" | "votes_count" | "users_count",
+  dates: Date[],
 ) {
-  const analyticsByDay = await Promise.all(
-    metrics.map((metric) => getAnalyticsMetricListPerDay(metric, options)),
-  );
-
-  return analyticsByDay;
+  const keys = dates.map((date) => [metric, formatDate(date)]);
+  const res = await kv.getMany<bigint[]>(keys);
```

Comment:
**praise:** nice to see the refactoring to `.getMany`.  
this was also one of the first approaches i had in mind when starting w/ #289.  

**question:** would it make sense to check `dates.length` so we don't run in any issues w/ the limit?  
if we want to continue that route, would it also make sense to split `dates` into chunks of 10s?

Reply:
`kv.getMany()` will throw if the limit is exceeded anyway, so I think it's fine in its current state. If needed, we can work on "batching" this function up in another PR.